lib/auth/join_ec2.go

```diff
@@ -223,59 +218,46 @@ func kubeExists(ctx context.Context, presence services.Presence, hostID string)
 }
 
 func appExists(ctx context.Context, presence services.Presence, hostID string) (bool, error) {
-	namespaces, err := presence.GetNamespaces()
+	apps, err := presence.GetApplicationServers(ctx, defaults.Namespace)
```

Comment:
Could we query for the specific app?

Same for databases and okta below.

Reply:
Thanks for looking. It's silly that this enumerates resources to find a specific one we know about, but it's also pre-existing behavior.